<?xml version="1.0" encoding="utf-8"?>

<robot xmlns:xacro="http://www.ros.org/wiki/xacro">  
  <xacro:macro name="solid_wood_block" params="handle parent position orientation">

  <joint name="solid_wood_block_${handle}_joint" type="fixed">
    <origin xyz="${position}" rpy="${orientation}"/>
    <parent link="${parent}"/>
    <child link="solid_wood_block_${handle}"/>
  </joint>

    <link
      name="solid_wood_block_${handle}">
      
      <inertial>
        <origin xyz="0 0 0.5" rpy="0 0 0"/>
        <mass value="1"/>
        <inertia
      ixx="1.0" ixy="0.0" ixz="0.0"
      iyy="1.0" iyz="0.0"
      izz="1.0"/>
      </inertial>
      
      <visual>
        <origin
          xyz="0 0 0"
          rpy="0 0 0" />
        <geometry>
          <mesh
            filename="package://ICRA2023_Quadruped_Competition/meshes/solid_wood_block.dae"/>
        </geometry>
        <material
          name="">
          <color
            rgba="0.22 0.76 0.94 1" />
          <texture filename="package://ICRA2023_Quadruped_Competition/textures/wooden.jpg" />
        </material>
      </visual>
      <collision>
        <origin
          xyz="0 0 0"
          rpy="0 0 0" />
        <geometry>
          <mesh
            filename="package://ICRA2023_Quadruped_Competition/meshes/solid_wood_block.dae" />
        </geometry>
      </collision>
    </link>
  
  </xacro:macro>
</robot>